### PR TITLE
12.0 csv export improvements

### PR DESCRIPTION
**Pull request by victor-champonnois**
## Description
- remove notes and sections from csv invoice export
- take discount into account when computing the total_amount

## Odoo task (if applicable)
https://gestion.coopiteasy.be/web#id=9389&action=475&active_id=198&model=project.task&view_type=form&menu_id=536

## Checklist before approval

- [ ] Tests are present (or not needed).
- [ ] Credits/copyright have been changed correctly.
- [ ] Change log snippet is present.
- [ ] (If a new module) Moving this to OCA has been considered.

### CODE REVIEW

File: csv_export_invoice/models/csv_export_invoice.py [hunk 1]
```diff
@@ -127,16 +127,18 @@ def export_order(invoice_line):
...
+        :param line: account.invoice.line record
         :return: a tuple of total amount, base_amount and tax_amount.
           base_amount and tax_amount are rounded to two decimals.
           total_amount is the sum of rounded base_amount and tax_amount
           for lines in the invoice.
         """
         taxes = line.invoice_line_tax_ids
+        # from src/odoo/addons/account/models/account_invoice.py
+        # is there a way not to copy paste this code ?
+        price = line.price_unit * (1 - (line.discount or 0.0) / 100.0)
```
Comment (robinkeunen) [inline csv_export_invoice/models/csv_export_invoice.py]: I had to do this in v9 to make sur the taxes were localized (ie. a 21% tax on a product converts to 0% intracomm in the case of a sale in EU).

Maybe the fields (`invoice_line_tax_ids` ?) in the account.invoice.line are better in v12. You'd need to look at function `_compute_price` which seems to take the discounts and taxes into account when setting

- price_subtotal
- price_total
- price_subtotal_signed

**Review by robinkeunen — APPROVED** [review]
Looks good, you could take a closer look at the accounting in v12 to _eventuellement_ simplify my code. But this is not strictly necessary.

File: csv_export_invoice/models/csv_export_invoice.py (hunk 1 above)
Comment (victor-champonnois) [reply csv_export_invoice/models/csv_export_invoice.py]: Not sure we understood each other, the comment was about the line `price = line.price_unit * (1 - (line.discount or 0.0) / 100.0)` that I copied from the _compute_price method :

https://github.com/OCA/OCB/blob/daa274eb48324d563e1096e334a5a73034be71dc/addons/account/models/account_invoice.py#L1695

Comment (robinkeunen) [reply csv_export_invoice/models/csv_export_invoice.py]: Yes and the rest of the code is copied from there as well.

**Comment by victor-champonnois** [comment]
@robinkeunen OK, looking at it I don't see an easy refactor. Let's keep it like this with a comment to explicit the fact that it's copy/paste

**Comment by robinkeunen** [comment]
@victor-champonnois 👍

**Comment by victor-champonnois** [comment]
@robinkeunen refactored to simplify, can you check  ?

**Review by robinkeunen — APPROVED** [review]
LGTMn much simpler. It's worth notifying Jérémie of the refactor so that he can have a closer look at the exports from the test server.

**Pull request merged**